#! /bin/sh /usr/share/dpatch/dpatch-run
## 01configure_fix_alpha_vsprintf.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Enhance the vsprintf() test to also succeed on Alpha.
## DP: Cf. #213842.

@DPATCH@
diff -Naur screen-4.0.3.orig/configure.in screen-4.0.3/configure.in
--- screen-4.0.3.orig/configure.in	2008-06-01 23:39:48.000000000 +0200
+++ screen-4.0.3/configure.in	2008-06-01 23:39:51.000000000 +0200
@@ -1210,7 +1210,7 @@
 rm -f /tmp/conftest*
 
 AC_MSG_CHECKING(for vsprintf)
-AC_TRY_LINK(,[vsprintf(0,0,0);], AC_MSG_RESULT(yes);AC_DEFINE(USEVARARGS), AC_MSG_RESULT(no))
+AC_TRY_LINK([#include <stdarg.h>],[va_list valist; vsprintf(0,0,valist);], AC_MSG_RESULT(yes);AC_DEFINE(USEVARARGS), AC_MSG_RESULT(no))
 
 AC_HEADER_DIRENT
 
